### ISSUE TITLE
Forward search inside include/input files

### ISSUE THREAD

**Issue by tiagovla** [latex-lsp/texlab#863]
This is a follow up of https://github.com/latex-lsp/texlab/discussions/862#discussion-4927487.

I couldn't find it, but is there a way to mark the root file path in an included file so it finds the correct pdf path?
Maybe a comment like ``%TexLab_ROOT=../main.tex``.

## Expected behavior
```bash
zathura --synctex-forward 330:1:$ROOT_PATH/subfolder/file.tex $ROOT_PATH/main.pdf
```

## Current behavior
```bash
zathura --synctex-forward 330:1:$ROOT_PATH/subfolder/file.tex $ROOT_PATH/subfolder/file.pdf
```

I'm currently using neovim, as I can see, the [lsp request](https://github.com/neovim/nvim-lspconfig/blob/2ba6e268d92934e14fe561fe446a49540803e57f/lua/lspconfig/server_configurations/texlab.lua#L38-L41) uses the same `textDocument uri` for both paths (.tex and .pdf), If it took another parameter (and defaulted to `textDocument uri`), I could reuse texlab's logic and locally implement the comment solution myself.

**Comment by pfoerster** [comment]
@tiagovla

> This is a follow up of https://github.com/latex-lsp/texlab/discussions/862#discussion-4927487.

Normally, this should already work out of the box. Using forward search from a child document should use the parent's PDF. Do you have `texlab.rootDirectory` set to some value somehow? If not, a log file (use `texlab -vvvv --log-file /tmp/texlab.log`) might be helpful.

Regarding the project detection: There is a wiki page which describes the algorithm.
*TLDR*: You can add an empty `.texlabroot` file in the same directory as `main.tex` and the server uses this directory as the root directory.

**Comment by tiagovla** [comment]
I will try to isolate it and create a reproducible example with logs. Thanks.

**Comment by pfoerster** [comment]
@tiagovla
Version 5.4.0 now ships with a workspace command (see [`texlab.showDependencyGraph`](https://github.com/latex-lsp/texlab/wiki/Workspace-commands#texlabshowdependencygraph)) that could be helpful here. You can render the result of this command with GraphViz.

**Comment by tiagovla** [comment]
I got some free time to dig into it today, but after updating texlab, this problem is completely gone. Everything works as it should, including in a project that wouldn't work before.
